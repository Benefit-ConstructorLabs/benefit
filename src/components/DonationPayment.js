import React from 'react';

import PaymentDetailsFormContainer from '../containers/PaymentDetailsFormContainer';
import PaymentAcknowledgeContainer from '../containers/PaymentAcknowlegeContainer';

const DonationPayment = ({ donationComplete }) => (
  <React.Fragment>
    {!donationComplete
      && (
        <PaymentDetailsFormContainer />
      )
    }
    {donationComplete
      && (
        <PaymentAcknowledgeContainer />
      )
    }
  </React.Fragment>
);

export default DonationPayment;
