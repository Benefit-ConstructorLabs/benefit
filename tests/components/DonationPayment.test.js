import React from 'react';
import renderer from 'react-test-renderer';
import DonationPayment from '../../src/components/DonationPayment';

describe('DonationPayment', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<DonationPayment />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
