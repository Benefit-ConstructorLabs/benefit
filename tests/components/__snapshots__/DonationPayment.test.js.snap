// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationPayment should render correctly 1`] = `
Array [
  <h2>
    Donation Payment
  </h2>,
  <div
    className="payment-details-form"
  >
    <h3>
      Set up payment
    </h3>
    <form
      className="payment-details-form__form"
    >
      <input
        placeholder="Card number"
        type="text"
      />
      <input
        placeholder="Expiry date"
        type="text"
      />
      <input
        placeholder="CCV"
        type="text"
      />
      <p>
        Set up a donor account?
      </p>
      <p>
        Continue donating anonymously:
      </p>
      <button
        type="button"
      >
        Donate -Price- to -First Name-
      </button>
    </form>
  </div>,
  <section
    className="acknowledgement"
  >
    <p
      className="acknowledgement__notification"
    >
      Donation confirmed
    </p>
    <h2
      className="acknowledgement__confirmation"
    >
      All set. Thanks for helping “ FirstName” out!
    </h2>
    <h3
      className="acknowledgement__question"
    >
      Want to save your details for another donation?
    </h3>
    <button
      className="btn acknowledgement__account__btn"
      onClick={[Function]}
      type="button"
    >
      Set up donor account
    </button>
  </section>,
]
`;
